src/memory.cc

```diff
@@ -97,14 +97,20 @@ void* operator new(size_t size) BADALLOC
 		}
 		else
 		{
+#if defined(_CXXRT_NO_EXCEPTIONS)
+			break;
+#else
 			throw std::bad_alloc();
+#endif
 		}
 		mem = malloc(size);
 	}
 
 	return mem;
 }
 
+
+#if !defined(_CXXRT_NO_EXCEPTIONS)
```

Comment:
I think we still want to provide the nothrow versions for the noexcept case because this is the one that does provide an API contract that allows error checking.

This is documented in the standard to call the other one, but we can do this by setting the new handler for the thread to a function that we use to indicate failure and resetting it on return.

Reply:
Added back these functions, delegating to the other one for now as discussed offline. Happy to revisit this later with a thread-local new handler